test/unit/math/binaryOps/src/binaryOps.cpp

```diff
@@ -0,0 +1,432 @@
+template<
+        typename TAcc,
+        typename T>
+using alpaka_func_ptr = T (*) (TAcc const & , T const &, T const &);
+
+template<
+        typename T>
+using std_func_ptr = T (*) (T, T);
+
+// used for all operators that need const references (like min())
+template<
+        typename T>
+using std_func_ptr_const = T const & (*) (T const & , T const &);
+
+
+template<
+        typename TAcc,
+        typename T>
+struct TestStruct
+{
+    std_func_ptr<T> stdOp ;
+    std_func_ptr_const <T> stdAlternative;
+    alpaka_func_ptr<TAcc, T> alpakaOp;
+    Range range;
+};
+
+
+class BinaryOpsKernel{
+public:
+    ALPAKA_NO_HOST_ACC_WARNING
+    template<
+            typename TAcc,
+            typename TData,
+            typename TIdx
+    >
+    ALPAKA_FN_ACC auto operator()(
+            TAcc const & acc,
+            TIdx const & numOps,
+            TIdx const & sizeArgs,
+            TData const * const argsX,
+            TData const * const argsY,
+            TestStruct<TAcc, TData> const * const structs,
+            TData * results
+    ) const
+    -> void
+    {
+        auto const gridThreadIdx(alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u]);
+        TData argX;
+        TData argY;
+        if(gridThreadIdx < numOps)
+        {
+            // sizeRes = numOps * sizeArgs
+            switch (structs[gridThreadIdx].range)
+            {
+                case Range::POSITIVE_ONLY:
+                    for(TIdx i(0); i < sizeArgs/2 -1; ++i)
+                    {
+                        argX = argsX[i];
+                        argY = argsY[i];
+                        results[i + gridThreadIdx* sizeArgs] = structs[gridThreadIdx].alpakaOp(acc, argX, argY);
+                    }
+                    break;
+                case Range::NOT_ZER0:
+                    for(TIdx i(0); i < sizeArgs; ++i)
+                    {
+                        if(i == sizeArgs/2 -1 || i == sizeArgs/2)
+                            continue;
+                        argX = argsX[i];
+                        argY = argsY[i];
+                        results[i + gridThreadIdx* sizeArgs] = structs[gridThreadIdx].alpakaOp(acc, argX, argY);
+                    }
+                    break;
+
+                case Range::X_NOT_ZERO:
+                    for(TIdx i(0); i < sizeArgs/2 -1; ++i)
+                    {
+                        if(i == sizeArgs/2 -1 || i == sizeArgs/2)
+                            argX = argsX[0];
+                        else
+                            argX = argsX[i];
+                        argY = argsY[i];
+                        results[i + gridThreadIdx* sizeArgs] = structs[gridThreadIdx].alpakaOp(acc, argX, argY);
+                    }
+                    break;
+
+                case Range::Y_NOT_ZERO:
+                    for(TIdx i(0); i < sizeArgs/2 -1; ++i)
+                    {
+                        if(i == sizeArgs/2 -1 || i == sizeArgs/2)
```

Comment:
`i == sizeArgs / 2 - 1 || i == sizeArgs / 2` will always be `false` because the loop condition to not allow this values for `i`

Reply:
same for `X_NOT_ZERO`